assignments/array-methods.js

```diff
@@ -54,30 +54,62 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
+let allCaps = runners.map(person => person.first_name.toUpperCase());
+console.log(allCaps);
 
 // ==== Challenge 3: Use .filter() ====
-// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+// The large shirts won't be available for the event due to an ordering issue.
+// Get a list of runners with large sized shirts so they can choose a different size.
+// Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
+let largeShirts = runners.filter(person => person.shirt_size === "L");
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
-// The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+// The donations need to be tallied up and reported for tax purposes.
+// Add up all the donations into a ticketPriceTotal array and log the result
+let ticketPriceTotal = runners.reduce((acc, next) => acc + next.donation, 0);
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
-// Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to solve 3 unique problems using one or many of the array methods listed above.
+// Now that you have used .forEach(), .map(), .filter(), and .reduce().
+// I want you to think of potential problems you could solve given the data set and the 5k fun run theme.
+// Try to solve 3 unique problems using one or many of the array methods listed above.
 
-// Problem 1
+// Problem 1: Project an array of individuals who have donated greater than 200
+// the projected array must have only 2 fields - a fullName field and a donation field
+let greaterDonors = runners
+  .filter(person => person.donation > 200)
+  .map(person => ({
+    fullName: `${person.first_name} ${person.last_name}`,
+    donation: person.donation
+  }));
+console.log(greaterDonors);
 
-// Problem 2
+// Problem 2: Find the person who has the highest donation using .reduce() method
+let mostGenerousDonor = runners.reduce((acc, next) => {
+  if (next.donation > acc.donation) return next;
+  return acc;
+});
+console.log(mostGenerousDonor.first_name);
 
-// Problem 3
\ No newline at end of file
+// Problem 3: Using the .reduce() method create an object that contains the number of shirts to order for each size
+// The output object must contain { "M": <number of M>, "L": <number of L> ... }
+let shirtOrders = runners.reduce(
+  (acc, next) => ({
+    ...acc,
+    [next.shirt_size]: (acc[next.shirt_size] || 0) + 1
+  }),
+  {}
+);
+console.log(shirtOrders);
```

Comment:
All of these are really cool. I love that you showed off some of the cool things we can do with ES6 syntax. Reduce is crazy powerful and there are so many different ways that it is used and implemented. It's pretty crazy. Awesome job here.